packages/firestore/src/local/memory_persistence.ts

```diff
@@ -211,6 +318,7 @@ export class MemoryLruDelegate implements ReferenceDelegate, LruDelegate {
   ): PersistencePromise<number> {
     let count = 0;
     const cache = this.persistence.getRemoteDocumentCache();
+    // TODO(gsoltis): make this concurrent
```

Comment:
Hm. This is an in-memory data structure. I would be surprised if this could be sped up if we allowed out-of-order execution.